### DISCUSSION TITLE
[Discussion thread] Smart charging

### DISCUSSION THREAD

**Discussion by nopeless** [seerge/g-helper#1409] [General]
2023 G14 came with the ability to interact with the Windows smart charging api

https://support.microsoft.com/en-us/windows/use-smart-charging-in-windows-2ac1b4ba-6027-410a-b80e-f6767d867002

This provides an additional visual indicator 

<img width="601" alt="Surface-smart-charging-icon-2-1690555490" src="https://github.com/seerge/g-helper/assets/38830903/13099fbc-db8e-4644-8f20-84179e87b395">

On a clean windows install, however, this doesn't work/exist. I am not sure why this is the case, so I ask for clues

**Comment by seerge** [comment]
@nopeless hello,

This is something Asus has added in 2023 models only as far as I have noticed (when tested it on G14 4080)

There is some new low level notification inside ACPI call related to setting charge limit, that apparently can communicate with windows and let it know that battery is "smart charged" when limit is <= 80 or something like that.

But I don't know how to send it manually :) (and if it's even possible)

**Reply by nopeless** [reply]
Ok, at least you acknowledge. I'll wait for people to catch on to this feature and perhaps find how it works. Afaik, this is identical to what MyAsus/Ghelper has been doing all along, just a visual difference to the battery icon.

I would say we should just leave this thread open for now

**Reply by seerge** [reply]
@nopeless this is a code from decompiled DSDT file responsible for setting charge limit

```
                If ((IIA0 == 0x00120057))
                {
                    If ((IIA1 <= 0x50))
                    {
                        ^^PCI0.SBRG.MBF4 = 0x50
                        Notify (^^PCI0.SBRG.BAT0, 0x80) // Status Change
                    }
                    Else
                    {
                        ^^PCI0.SBRG.MBF4 = Zero
                        Notify (^^PCI0.SBRG.BAT0, 0x80) // Status Change
                    }

                    Return (SRSC (IIA1))
                }
```
``Notify `` thing kind of tips that there is some communication happening between bios handling charging and windows.

**Comment by AudreyAP** [comment]
My M16 2023 has the "smart charge" icon working after a clean-install, even with zero networking / no driver updates installed whatsoever, just absolutely stock win11 iso.

**Reply by nopeless** [reply]
Thanks for letting me know. That wasn't the case for my g14 2023

**Reply by AudreyAP** [reply]
@nopeless I just double checked, it only shows up if the charge limit is set to 80% or less. 95% charge limit etc does not show the icon. Maybe that's it?

**Reply by nopeless** [reply]
Oh what?

**Reply by nopeless** [reply]
That makes no sense but apparently that is the case. My mind is blown @seerge @AudreyAP

**Reply by AudreyAP** [reply]
@nopeless I think it's because the stock battery limiter is just an 80% on/off toggle. Weird.

**Comment by Kikimahe** [comment]
Interesting.

Could this visual indicator be displayed on all ASUS models by Ghelper when the charging limit is enabled in Ghelper ? :-)

**Comment by iatacs19** [comment]
Just to add another point of data: ProArt Studiobook 16 OLED H7604JI-DS96T has this indicator as well. I did a clean install without any Asus services.

**Comment by Stoni85** [comment]
This might be already discussed or solved meantime: I seems I have smart charged enabled ant 80% max charge, as well as g-helper set the same way at 80% max and they both work when in on mode, and sleeping mode. Whenever I turn the laptop off, it will go charging to 100%.  It's 2025 nad I have a g14 2024 and it seems there's still no solutions, if not installing myasus.
